src/snowflake/connector/network.py

```diff
@@ -268,8 +268,16 @@ def get_connection(
                 )
             proxy_manager = self.proxy_manager_for(proxy)
 
-            # Add Host to proxy header SNOW-232777
-            proxy_manager.proxy_headers["Host"] = parsed_url.hostname
+            try:
```

Comment:
If `SOCKSProxyManager` is the only case this fails, is it better if we just if-else instead of using a try catch?

Reply:
good question, the way I made the change is because `SOCKSProxyManager` is dynamically resolved:

https://github.com/snowflakedb/snowflake-connector-python/blob/main/src/snowflake/connector/vendored/requests/adapters.py#L56-L61

```python
try:
    from ..urllib3.contrib.socks import SOCKSProxyManager
except ImportError:

    def SOCKSProxyManager(*args, **kwargs):
        raise InvalidSchema("Missing dependencies for SOCKS support.")
```

if `pysocks` is installed,  `SOCKSProxyManager` is a class, however if `pysocks` is not installed, `SOCKSProxyManager` is a function in which case we can not do the instance check `if instance(proxy_manager, SOCKSProxyManager)` which will raise `TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union`.

so I avoid importing `SOCKSProxyManager`, otherwise I need to handle the complexity of importing errors.

Reply:
What about checking whether the return type is an instance of `ProxyManager` and only set `proxy_headers` if so? I don't have a preference tho.

Reply:
yep I think this would work, code updated!